geopandas/geodataframe.py

```diff
@@ -48,6 +48,8 @@ class GeoDataFrame(GeoPandasBase, DataFrame):
     geometry : str or array (optional)
         If str, column to use as geometry. If array, will be set as 'geometry'
         column on GeoDataFrame.
+    sindex_cls : spatial index implementation (optional)
+        A class that implements a spatial index respecting the API of geopandas.sindex
```

Comment:
Is this needed for this PR? 
(this is certainly something to discuss, but for now, would rather try to limit it to the minimal needed to start using pygeos instead of rtree)

Reply:
I suppose this is for being able both rtree sindex and pygeos sindex (even when using pygeos under the hood)? 
I think on the short term, that can also be done by directly setting the `_sindex` attribute (that should suffice for testing by power users). 

(and again, longer term, we should certainly discuss how to be more flexible about which spatial index class is used through an official API)

Reply:
As you indicate, this is _not_ really intended for this PR (I don't even think I wrote tests for this?). It was more just because it was brought up in #1344 and I was just exploring the idea here. I will remove this when I clean up this PR.